docs/descriptor.rst

```diff
@@ -0,0 +1,210 @@
+          value: "JBoss EAP 7.0"
+
+Environment variables
+---------------------
+
+Similar to labels -- we can specify environment variables that should be
+present in the container after running the image. We provide ``envs``
+section for this.
+
+Environment variables can be divided into two types:
+
+1. **Information environment variables** -- these are set and available in
+   the image. This type of envirpnment variables provide information to
+   the image consumer. In most cases such environment variables *should not*
+   be modified.
+2. **Configuration environment variables** -- this type of variables are
+   just a hint to the image consumer that such environment variables can be
+   set. These environment variables are **not** set during image build time.
+   Every **configuration** enviromnent variable should provide an example usage
+   (``example``) and short description (``description``).
+
+Please note that you could have an environment variable with both: a ``value``
+and ``example`` set. This suggest that this environment variable could be redefined.
+
+.. note::
+
+    Configuration environment variables (without ``value``) are not
+    generated to the build source. These can be used instead as a
+    source for generating documentation.
+
+.. code:: yaml
+
+    envs:
+        - name: "STI_BUILDER"
+          value: "jee"
+        - name: "JBOSS_MODULES_SYSTEM_PKGS"
+          value: "org.jboss.logmanager,jdk.nashorn.api"
+        - name: "OPENSHIFT_KUBE_PING_NAMESPACE"
+          example: "myproject"
+          description: "Clustering project namespace."
+        - name: "OPENSHIFT_KUBE_PING_LABELS"
+          example: "application=eap-app"
+          description: "Clustering labels selector."
+
+Ports
+-----
+
+This section is used to mark which ports should be exposed in the
+container. If we want to highlight a port used in the container, but not necessary expose
+it -- we should set the ``expose`` flag to ``false`` (``true`` by default).
+
+.. code:: yaml
+
+    ports:
+        - value: 8443
+        - value: 8778
+          expose: false
+
+Command and entrypoint
+----------------------
+
+You can specify the entrypoint or command that should be used by the
+container with the ``cmd`` and ``entrypoint``.
+
+.. note::
+
+    Both ``entrypoint`` and ``cmd`` keys use the array form of
+    providing its value.
+
+.. code:: yaml
+
+    entrypoint:
+        - "/opt/eap/bin/wrapper.sh"
+    cmd:
+        - "some cmd"
+        - "argument"
+
+Packages
+--------
+
+If you need to install additional packages you can use the ``packages``
+section where you specify package names to be installed.
+
+.. todo::
+
+    Adding repo files
+
+.. code:: yaml
+
+    packages:
+        - mongodb24-mongo-java-driver
+        - postgresql-jdbc
+        - mysql-connector-java
+        - maven
+        - hostname
+
+Artifacts
+---------
+
+It's not a common case where your image won't require external artifacts.
```

Comment:
Its common for images to require external artifacts - sounds clearer than double negation for me.

Reply:
I'll change it.